judge/runpipe.cc

```diff
@@ -846,6 +913,18 @@ struct state_t {
           }
           continue;
         }
+        if (fd == child_timelimit_pipe) {
+            static char buffer[1];
+            if (read(child_timelimit_pipe, buffer, 1) != 1) {
+                if (errno != EAGAIN && errno != EWOULDBLOCK) {
+                    error(errno, "failed to read from tle pipe");
+                }
```

Comment:
I'm not sure that the fall-through behaviour when `read` has an error (or returns zero) but the error is not `EAGAIN` or `EWOULDBLOCK` is correct here: in the original code for `child_exited_pipe` it would then check if a child exited and not do anything if not, but here `child_indicated_timelimit = true` is automatically set.

Reply:
Done